pkg/dns/client/proxy.go

```diff
@@ -45,24 +48,33 @@ func newDNSProxy(protocol string, resolver *LocalDNSServer) (*dnsProxy, error) {
 		protocol: protocol,
 		resolver: resolver,
 	}
+	addr := resolver.addr
+	if addr == "" {
+		addr = "localhost:15053"
+	}
+	if strings.HasSuffix(addr, ":53") && os.Getuid() != 0 {
+		log.Error("DNS address :53 and not running as root, use default")
```

Comment:
nit: isn't it technically possible to do this without root by giving the bind_serve cap?